packages/core/src/exchanges/ssr.ts

```diff
@@ -188,7 +197,8 @@ const revalidated = new Set<number>();
 export const ssrExchange = (params: SSRExchangeParams = {}): SSRExchange => {
   const staleWhileRevalidate = !!params.staleWhileRevalidate;
   const includeExtensions = !!params.includeExtensions;
-  const data: Record<string, SerializedResult | null> = {};
+  const data: SSRDataStorage =
```

Comment:
Why can't we do this with initialState? For instance when we run extractData we extract into this custom storage and on the client we can leverage the restoreData method and initialState to restore additional and the initial entries.

This is similar to our nextjs solution